src/test/java/com/iexec/core/task/update/TaskUpdateRequestManagerTests.java

```diff
@@ -105,4 +147,97 @@ public void shouldNotUpdateAtTheSameTime() throws NoSuchFieldException, IllegalA
+                List.of(
+                        initializingTask,
+                        completedTask,
+                        runningTask,
+                        initializedTask,
+                        consensusReachedTask
+                )
+        );
+        Collections.shuffle(tasks);
+        queue.addAll(tasks);
+
+        final List<TaskUpdate> prioritizedTasks = queue.takeAll();
+        Assertions.assertThat(prioritizedTasks)
+                .containsExactly(
+                        completedTask,
+                        consensusReachedTask,
+                        runningTask,
+                        initializedTask,
+                        initializingTask
+                );
+    }
+
+    @Test
+    void shouldGetInOrderForContributionDeadline() throws InterruptedException {
+        final TaskUpdatePriorityBlockingQueue queue = taskUpdateRequestManager.queue;
+
+        final Date d1 = new GregorianCalendar(2021, Calendar.JANUARY, 1).getTime();
+        final Date d2 = new GregorianCalendar(2021, Calendar.JANUARY, 2).getTime();
+        final Date d3 = new GregorianCalendar(2021, Calendar.JANUARY, 3).getTime();
+        final Date d4 = new GregorianCalendar(2021, Calendar.JANUARY, 4).getTime();
+        final Date d5 = new GregorianCalendar(2021, Calendar.JANUARY, 5).getTime();
+
+        TaskUpdate t1 = new TaskUpdate(Task.builder().currentStatus(TaskStatus.RUNNING).contributionDeadline(d1).build(), null, null);
+        TaskUpdate t2 = new TaskUpdate(Task.builder().currentStatus(TaskStatus.RUNNING).contributionDeadline(d2).build(), null, null);
+        TaskUpdate t3 = new TaskUpdate(Task.builder().currentStatus(TaskStatus.RUNNING).contributionDeadline(d3).build(), null, null);
+        TaskUpdate t4 = new TaskUpdate(Task.builder().currentStatus(TaskStatus.RUNNING).contributionDeadline(d4).build(), null, null);
+        TaskUpdate t5 = new TaskUpdate(Task.builder().currentStatus(TaskStatus.RUNNING).contributionDeadline(d5).build(), null, null);
+
+        List<TaskUpdate> tasks = new ArrayList<>(List.of(t1, t2, t3, t4, t5));
+        Collections.shuffle(tasks);
+        queue.addAll(tasks);
+
+        final List<TaskUpdate> prioritizedTasks = queue.takeAll();
+        Assertions.assertThat(prioritizedTasks)
+                .containsExactly(
+                        t1,
+                        t2,
+                        t3,
+                        t4,
+                        t5
+                );
+    }
+
+    @Test
+    void shouldGetInOrderForStatusAndContributionDeadline() throws InterruptedException {
+        final TaskUpdatePriorityBlockingQueue queue = taskUpdateRequestManager.queue;
+
+        final Date d1 = new GregorianCalendar(2021, Calendar.JANUARY, 1).getTime();
+        final Date d2 = new GregorianCalendar(2021, Calendar.JANUARY, 2).getTime();
+
+        TaskUpdate t1 = new TaskUpdate(Task.builder().currentStatus(TaskStatus.RUNNING).contributionDeadline(d1).build(), null, null);
```

Comment:
`buildTaskUpdate(TaskStatus status, Date contributionDeadline)` in yours tests would be cleaner

Reply:
Right. But we have some `TaskUpdate` with `chainTaskId`, `locks` or `consumer` so we need to include them as well to our new method. That's done in https://github.com/iExecBlockchainComputing/iexec-core/pull/479/commits/3c29ec47e8ecb2ae0165d3c1927ef199f76c5d1b.